plugins/volumeFilter_test.go

```diff
@@ -0,0 +1,439 @@
+				config:                 &VolumeFilterConfig{additionalMarketIDs: marketIDs},
+				dailyVolumeByDateQuery: marketConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - account ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{optionalAccountIDs: accountIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs},
+				dailyVolumeByDateQuery: accountConfigVolumeQuery,
+			},
+		},
+		{
+			name:           "success - account and market ids",
+			configValue:    "",
+			exchangeName:   "",
+			tradingPair:    &model.TradingPair{Base: "XLM", Quote: "XLM"},
+			assetDisplayFn: testAssetDisplayFn,
+			baseAsset:      testNativeAsset,
+			quoteAsset:     testNativeAsset,
+			config:         &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+			db:             db,
+			wantError:      nil,
+			wantSubmitFilter: &volumeFilter{
+				name:                   "volumeFilter",
+				baseAsset:              testNativeAsset,
+				quoteAsset:             testNativeAsset,
+				config:                 &VolumeFilterConfig{optionalAccountIDs: accountIDs, additionalMarketIDs: marketIDs},
+				dailyVolumeByDateQuery: accountMarketConfigVolumeQuery,
+			},
+		},
+	}
+
+	for _, k := range testCases {
+		t.Run(k.name, func(t *testing.T) {
+			actual, e := makeFilterVolume(
+				k.configValue,
+				k.exchangeName,
+				k.tradingPair,
+				k.assetDisplayFn,
+				k.baseAsset,
+				k.quoteAsset,
+				k.db,
+				k.config,
+			)
+
+			assert.Equal(t, k.wantError, e)
+			assert.Equal(t, k.wantSubmitFilter, actual)
+		})
+	}
+}
+
+func TestVolumeFilterFn(t *testing.T) {
+	db := connectTestDb()
+	dailyVolumeByDateQuery, e := makeDailyVolumeByDateQuery(db, "", "native", "native", []string{}, []string{})
+	if e != nil {
+		t.Log("could not make empty config volume query")
+		return
+	}
+
+	emptyFilter := &volumeFilter{
+		name:                   "volumeFilter",
+		configValue:            "",
+		baseAsset:              testNativeAsset,
+		quoteAsset:             testNativeAsset,
+		config:                 &VolumeFilterConfig{},
+		dailyVolumeByDateQuery: dailyVolumeByDateQuery,
+	}
+
+	testCases := []struct {
+		name      string
+		filter    *volumeFilter
+		dailyOTB  *VolumeFilterConfig
+		dailyTBB  *VolumeFilterConfig
+		op        *txnbuild.ManageSellOffer
+		wantOp    *txnbuild.ManageSellOffer
+		wantError error
+		wantTBB   *VolumeFilterConfig
+	}{
+		{
+			name:      "failure - is selling",
+			filter:    emptyFilter,
+			dailyOTB:  nil,
```

Comment:
a test case like this should be directly on `utils.IsSelling` (see reasons mentioned above).
can you add it there? (or put in a GH issue to be added later?)